CRM/Member/Form/Membership.php

```diff
@@ -1798,28 +1796,25 @@ protected function processContribution(
     $contributionParams
   ) {
     $contactID = $contributionParams['contact_id'];
-    $recurringContributionID = $this->legacyProcessRecurringContribution($params, $contactID);
-
-    if ($recurringContributionID) {
-      $contributionParams['contribution_recur_id'] = $recurringContributionID;
-    }
-
+    $contributionParams['contribution_recur_id'] = $this->legacyProcessRecurringContribution($params, $contactID);
     return CRM_Contribute_BAO_Contribution::add($contributionParams);
   }
 
   /**
-   * Create the recurring contribution record.
+   * Create the recurring contribution record if the form submission requires it.
    *
    * This function was copied from another form & needs cleanup.
    *
    * @param array $params
    * @param int $contactID
    *
-   * @return int
+   * @return int|null
    * @throws \CiviCRM_API3_Exception
    */
-  protected function legacyProcessRecurringContribution(array $params, $contactID): int {
-
+  protected function legacyProcessRecurringContribution(array $params, $contactID): ?int {
+    if (!$this->isCreateRecurringContribution()) {
```

Comment:
At the moment this check would never be true but with the check in place we can move the function call to a more generic place and it will only create if needed